src/client.ts

```diff
@@ -60,58 +67,64 @@ export function useGrid({
 	dependencies,
 }: {
 	getGrid: () => Promise<HGrid>
-	dependencies: ReadonlyArray<any>
+	dependencies: ReadonlyArray<unknown>
 }): GridResult {
-	const [grid, setGrid] = useState<HGrid>(new HGrid())
+	const data = useRef<GridData>()
 
-	const [isLoading, setIsLoading] = useState(true)
+	const gridData =
+		data.current ??
+		(data.current = { grid: new HGrid(), isLoading: true, loads: 0 })
```

Comment:
Is there a specific reason not to have this as the `initialValue` parameter of `useRef`?

Reply:
Everytime the hook is run, the grid object would be created unnecessarily even though we only need it created per instance. I thought this was a smarter way to handle this.

Reply:
Interesting! Makes sense, unfortunately `useRef` doesn't accept a lazy loading callback as `useState` does!

Reply:
Interesting! Makes sense, unfortunately `useRef` doesn't accept a lazy loading callback as `useState` does!

Reply:
This solution is also the officially suggested one: [https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily](https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily).